src/Paywall/index.tsx

```diff
@@ -0,0 +1,186 @@
+import type { Poool } from 'poool-access';
+import {
+  type ComponentPropsWithoutRef,
+  type MutableRefObject,
+  useEffect,
+  useMemo,
+  useRef,
+  useImperativeHandle,
+  forwardRef,
+} from 'react';
+import { classNames } from '@junipero/core';
+
+import type { AccessEvents, EventCallbackFunction } from '../types';
+import type { AccessContextValue } from '../contexts';
+import type { RestrictedContentRef } from '../RestrictedContent';
+import { generateId } from '../utils';
+import { useAccess } from '../hooks';
+
+export declare interface PaywallProps extends ComponentPropsWithoutRef<'div'> {
+  /**
+   * Ref to the content
+   */
+  contentRef?: MutableRefObject<RestrictedContentRef>;
+  /**
+   * Custom wrapper component ID
+   */
+  id?: string;
+  /**
+   * The paywall events
+   *
+   * More infos: https://www.poool.dev/docs/access/javascript/access/events
+   */
+  events?: AccessContextValue['events'];
+  /**
+   * The paywall config options
+   *
+   * More infos:
+   * https://www.poool.dev/docs/access/javascript/access/configuration
+   */
+  config?: AccessContextValue['config'];
+  /**
+   * The paywall texts options
+   *
+   * More infos: https://www.poool.dev/docs/access/javascript/access/texts
+   */
+  texts?: AccessContextValue['texts'];
```

Comment:
why you need to redefine each props of AccessContextValue and not `export declare interface PaywallProps extends Pick<AccessContextValue, 'events' | 'config' | 'texts'...>` ?